### ISSUE TITLE
Failing to build packages: Invalid version (double separator '-')

### ISSUE THREAD

**Issue by PSSGCSim** [zfsonlinux/spl#673]
Hello,
I have an issue while trying to build the spl deb/rpm packages. I am getting an error that says `error: line 68: Invalid version (double separator '-')`. I found same issue regarding this error under zfs itself https://github.com/zfsonlinux/zfs/issues/2369. But my error was not solved by removing all installed zfs/spl packages. I am trying to build from master under Linux `4.14.2-1` running a Debian. I am attaching the [build log](https://github.com/zfsonlinux/spl/files/1586241/build.log). My gcc is `gcc (Debian 7.2.0-18) 7.2.0` and RPM is `RPM version 4.14.0`.
Thank you in advance.

**Comment by ZeroPain** [comment]
i have the same problem on the newest ubuntu bionic or something what i did to fix it was go grab these packages from artful "prev version" and install them and it fixed my problem.

debugedit_4.12.0.2+dfsg1-2build2_amd64.deb
librpmbuild3_4.12.0.2+dfsg1-2build2_amd64.deb
librpmsign3_4.12.0.2+dfsg1-2build2_amd64.deb
rpm-common_4.12.0.2+dfsg1-2build2_amd64.deb
librpmbuild3_4.12.0.2+dfsg1-2build2_amd64.deb
rpm2cpio_4.12.0.2+dfsg1-2build2_amd64.deb
llibrpm3_4.12.0.2+dfsg1-2build2_amd64.deb
librpmio3_4.12.0.2+dfsg1-2build2_amd64.deb
rpm_4.12.0.2+dfsg1-2build2_amd64.deb

This is a issue with looks like rpm 4.14 and above don't know what the zfs guy can do about it only soultion other then manually building rpm and alien i could figure out. I hope this help anyone been bugging me for days

**Comment by PSSGCSim** [comment]
Yes, I tough it might be something in RPM itself but I was not sure. So I went upstream by just one step.
Maybe something needs to be adapted to match 4.14? But I do not know much about RPM.

But downgrading seems to work nicely. Thank you very much for suggesting it.

I went with these on Debian:

```
debugedit=4.12.0.2+dfsg1-2
librpm3=4.12.0.2+dfsg1-2
librpmbuild3=4.12.0.2+dfsg1-2
librpmbuild3=4.12.0.2+dfsg1-2
librpmio3=4.12.0.2+dfsg1-2
librpmsign3=4.12.0.2+dfsg1-2
rpm-common=4.12.0.2+dfsg1-2
rpm2cpio=4.12.0.2+dfsg1-2
rpm=4.12.0.2+dfsg1-2
```

**Comment by radiowest** [comment]
Still happening on Debian Testing as of 02/03/2018

**Comment by doubleddav** [comment]
I got the same thing on Ubuntu Artful, but got it all working in the end. The only thing I did differently was not use dkms.

I think the documentation here is incorrect: [https://github.com/zfsonlinux/zfs/wiki/Custom-Packages#kmod-rpm-and-deb](url)

After compiling spl and creating the debs, you're asked to install all of the generated debs. Install all of them EXCEPT spl-dkms_*.deb - dkms is for rpm distros, not Ubuntu.

Once you compile zfs and generate debs, install all of them EXCEPT zfs-dkms_*.deb
